File: app/src/main/java/com/example/cykashoppinglist/service/TodoService.java
package com.example.cykashoppinglist.service;

import com.example.cykashoppinglist.MainActivity;
import com.example.cykashoppinglist.R;
import com.example.cykashoppinglist.entity.GenericItem;
import com.example.cykashoppinglist.entity.Item;
import com.example.cykashoppinglist.mapper.TodoMapper;

public class TodoService implements RestService {
	private final String url;

	public TodoService() {
		this.url = MainActivity.mainContext.getResources().getString(R.string.host) + MainActivity.mainContext.getResources().getString(R.string.todoList);
	}

	@Override
	public void getAll() {
		doGet(url, TodoMapper.class);
	}

	@Override
	public void postItem(GenericItem item) {
		doPost(item, url, TodoMapper.class);
	}

	@Override
	public void deleteItem(Item item) {
		doDelete(item, url);
	}
}

File: app/src/main/java/com/example/cykashoppinglist/service/TodoService.kt
package com.example.cykashoppinglist.service

import com.example.cykashoppinglist.MainActivity
import com.example.cykashoppinglist.R
import com.example.cykashoppinglist.entity.GenericItem
import com.example.cykashoppinglist.entity.Item
import com.example.cykashoppinglist.mapper.TodoMapper

class TodoService : RestService {
	private val url: String

	override fun getAll() {
		doGet(url, TodoMapper)
	}

	override fun postItem(item: GenericItem?) {
		doPost(item!!, url, TodoMapper)
	}

	override fun deleteItem(item: Item?) {
		doDelete(item!!, url)
	}

	init {
		url = MainActivity.Companion.mainContext!!.getResources()
			.getString(R.string.host) + MainActivity.Companion.mainContext!!.getResources()
			.getString(R.string.todoList)
	}
}